### ISSUE TITLE
[Feature Request] Correct battery percentage

### ISSUE THREAD

**Issue by Pukson** [OnewheelCommunityEdition/OWCE_App#58]
Don't know if it does already, as I have a CBXR incoming. Onewheel app shows 0% even though we have enough juice left with the CBXR. 

Does this app calculate the percentage of the voltage? If not can that be an option. So we have a correct view of percentage if we use a bigger battery

**Comment by beeradmoore** [comment]
Hey @Pukson , the app currently uses what the board sends as battery % to display battery % so no go for CBXR on that front.

There is plans to add CBXR (and other battery mod) support. 
I believe the other branch [here](https://github.com/OnewheelCommunityEdition/OWCE_App/tree/renewed_squashedchanges_reporting3) has work to display percentage based on the current voltage. I need to review this however and then bring it into the main app. 

There is another enhancement that could be of use to you if it were added, #57

**Comment by Pukson** [comment]
Ah that sounds great! Can't wait to test it out

**Comment by Ryan526** [comment]
Is this still in development? About to install a quart in my pint and would be great to not have to consult a chart or install a chi gauge.

**Comment by beeradmoore** [comment]
Yep, just slowed. With everything else kicking up with OW currently progress is coming back and I hope I have something great to share soon.

**Comment by markysharky70** [comment]
@beeradmoore Hi...I'd love to help out by adding the JWXR battery mod voltage calculations for this app.  I noticed you have a branch in progress that adds the CBXR [here](https://github.com/OnewheelCommunityEdition/OWCE_App/tree/renewed_squashedchanges_reporting3) - just curious of that's a good branch to work from?  I'm figuring a settings slider with OFF/CBXR/JWXR settings would be the way to go.  But I'm a newb to this.  

Thanks.

**Comment by beeradmoore** [comment]
Hey @markysharky70 , that branch will be out of date currently so it will need to be brought into line with what is on development branch currently.

As far as I know (and correct me if I am wrong, I don't use battery mods) we don't need to pick battery type, we just need a toggle to change battery% display to be calculated from voltage rather than battery% reported directly from the board.

For both refining that algorithm as well as confirming it isn't going to be way out of whack when you are accelerating, regen, etc, from the voltage sag we put out an update which will allow people to record the raw data from their board and submit to OWCE API server. These rides can be re-played back through OWCE so you can develop without ever needing a real board. Based on those recordings we should be able to see how things behave with stock, CBXR, JWXR, etc and hopefully have the one size fits all system.

Problem is [I messed up](https://www.facebook.com/owceapp/posts/pfbid037rjEJa8VEwhskjfRVGK7NvZwpT3xh3Z8Zu6yrBSBgXKApT6cDUuMCbieiL8pCRfWl) and was not removing private data when the user selected it so I have gone and deleted all that data. I hope to get a new build soon so we can collect data again and then we can get back to work on using voltage as data source for battery%.

As of today I am not sure if there is work that can be done to help with this task directly. There is this other [thread](https://github.com/OnewheelCommunityEdition/OWCE_App/issues/59) which talks about other algorithms that can be used to calculate battery percentage if you want to familiarise yourself with that. If not, it will also be helpful to have an early-early-access tester who has an aftermarket battery to test out whatever gets built and confirm its working as expected.

**Comment by markysharky70** [comment]
Thanks for the info.  I’ll be happy to record my rides to provide data as well as beta test for you.  I am still trying to get up to speed on how to deploy to a device the IOS dev is new for me.
